sync rasterizers with engine

Tugsav opened the pull request:
The GLES3 rasterizer were calculating its own frame delta time. 
Apart from being a repetition of a calculation we are doing in the main script anyways, this meant that the frame_slice limit that is set in the main script were being ignored, leading to de-synchronization. This was visible when using particle systems since they need the frame delta time ().

This fix lets the rasterizer get the frame delta from the Engine singleton
That way any regulations to the frame step from the main script will not cause particle systems to process at a different step than the rest of the Engine.

Zylann commented:
Does it work if `VisualServer` runs in multithreaded mode?

Tugsav commented:
It ran without trouble in multi-threaded when i tried it. My thinking was that since the read is done in begin_frame() we can assume that we have to be in sync with the engine anyways, however it would be nice if someone a bit more knowledgeable about the rasterizer could confirm that.

reduz commented:
I'm not really sure this is the best way to do it, but I understand the need for it. I think a better way to do this is to add an argument to VisualServer::draw() and pass the elapsed time since the last frame (by default 0, as this is used from the editor too for stuff like onion skinning)

Tugsav committed:
Rasterizers are now in sync with engine

The rasterisers (both GLES3 and GLES2) were calculating their own frame delta time
This fix lets the rasterizers get the frame delta through the draw call
That way any regulations to the frame step from the main script will not cause particle systems to process at a different step than the rest of the Engine.

Remove unused rasterizer storage variable

frame.prev_tick variable were not used anywhere and has been removed

Tugsav commented:
@reduz I have attempted to implement a version of this the way you described. It works when i test it, however i have some additional questions about this part of the code that i noticed while implementing it:

The rasterizer storage `frame.prev_tick` variable is set in the `begin_frame` function that i have worked on in this PR. It does not however seem to be used anywhere (according to my limited grep skills) so i was wondering if i should just remove it instead of what i have done.
If not, should it be synced to the engine frame steps as well?

akien-mga commented:
> The rasterizer storage `frame.prev_tick` variable is set in the `begin_frame` function that i have worked on in this PR. It does not however seem to be used anywhere (according to my limited grep skills)

Confirmed:
```
$ rg prev_tick
drivers/gles2/rasterizer_storage_gles2.cpp
1992:   frame.prev_tick = 0;

drivers/gles2/rasterizer_gles2.h
46:     uint64_t prev_ticks;

drivers/gles2/rasterizer_gles2.cpp
233:    double delta = double(tick - prev_ticks) / 1000000.0;
243:    prev_ticks = tick;
256:    storage->frame.prev_tick = tick;
455:    prev_ticks = 0;

drivers/gles2/rasterizer_storage_gles2.h
817:            uint64_t prev_tick;

drivers/gles3/rasterizer_storage_gles3.h
1144:           uint64_t prev_ticks;
1185:                   prev_ticks = 0;
1420:           uint64_t prev_tick;

drivers/gles3/rasterizer_storage_gles3.cpp
5493:   particles->prev_ticks = 0;
5818:                   particles->prev_ticks = 0;
5836:                           particles->prev_ticks = 0;
7360:   frame.prev_tick = 0;

drivers/gles3/rasterizer_gles3.cpp
199:    double delta = double(tick - prev_ticks) / 1000000.0;
209:    prev_ticks = tick;
222:    storage->frame.prev_tick = tick;
454:    prev_ticks = 0;

drivers/gles3/rasterizer_gles3.h
47:     uint64_t prev_ticks;
```
(also copied in gles2 renderer but unused AFAICT, cc @karroffel)

karroffel commented:
Yep, I don't use the delta anywhere

mhilbrunner commented:
Can you fix the conflicts and squash the commits? Thanks :)

Tugsav commented:
Fixed conflicts and squashed :-)

hpvb commented:
Should this go in now?

akien-mga commented:
Sorry for the delay and thanks for your contribution!
